### DISCUSSION TITLE
How to assert for object exactly, except for one key by type

### DISCUSSION THREAD

**Discussion by jamischarles** [vitest-dev/vitest#5643] [Q&A]
Hi folks 👋 . LOVE vitest.

I have a very common situation where 90% of the values in my object are fixed, and 1 is variable (like a `time` value).

## What I've done in the past (but don't love):

1. `matchObject()` don't love this because it doesn't check for key completeness (you can add/remove some)
2. ^ plus I do a type check on the last value
3. populate the expected key with the real value

## What I'm looking for
Personally I'd LOVE something like this:
```js
   
const expectedEmitPayload = {
  "name": "joe",
  "type":"counter",
   "time": assert.isNumber, // always variable
} 


expect(mockEmitCb).toHaveBeenLastCalledWith(expectedEmitPayload);
```

Does anything like that exist? Thanks!

**Comment by sheremet-va** [comment]
Yes, it exists:

```ts
const expectedEmitPayload = {
  "name": "joe",
  "type":"counter",
   "time": expect.any(Number), // always variable
} 


expect(mockEmitCb).toHaveBeenLastCalledWith(expectedEmitPayload);
```
